sqlglot/dialects/hive.py

```diff
@@ -51,6 +51,13 @@
 DIFF_MONTH_SWITCH = ("YEAR", "QUARTER", "MONTH")
 
 
+def _create_sql(self, expression: exp.Create) -> str:
+    # remove Unique Column Constraints
+    for constraint in expression.find_all(exp.UniqueColumnConstraint):
+        if constraint.parent:
+            constraint.parent.pop()
+    return create_with_partitions_sql(self, expression)
```

Comment:
Can we just move this method inside Hive's `Generator` class? Also, can we merge its logic with Spark2's `_create_sql`? If that's possible then Spark will pick that up through inheritance.

Reply:
@GeorgeSittas I wanted to ask you this same question...
[Hive supports CREATE TEMPORARY TABLE AS SELECT](https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.0.1/using-hiveql/content/hive_create_a_hive_temporary_table.html) ... so moving the spark2 temporary table logic to hive would undo that.

I would be fine with moving the UNIQUE logic to spark2 and not worrying about that for Hive at this time. Thoughts?

Reply:
Ok no problem leaving Spark2 `_create_sql` as is then.

> I would be fine with moving the UNIQUE logic to spark2 and not worrying about that for Hive at this time. Thoughts?

Nah let's just keep it in Hive. Can we instead return `super(Hive.Generator, self).create_sql(expression)` for Spark2 in the fallback case where we don't find a temporary property? That way we'll reuse Hive's logic, but we need to first move `_create_sql` in Hive's generator and remove the `_` prefix to associate it with `exp.Create`.